gui/WorldView.cxx

```diff
@@ -1035,15 +1146,26 @@ void WorldView::setMesh(vtkSmartPointer<vtkPolyData> mesh)
 
   // Add this actor to the renderer
   d->renderer->AddActor(d->volumeActor);
-
-  if (mesh->GetPointData()->HasArray("RGB"))
+  bool mapScalars = false;
+  vtkDataArray* scalars = findArrayToColor(mesh, mapScalars);
+  if (scalars)
   {
-    meshMapper->ScalarVisibilityOn();
-    meshMapper->SelectColorArray("RGB");
+    if (mapScalars)
+    {
+      double range[2];
+      scalars->GetRange(range);
+      meshMapper->SetLookupTable(createBluesColorPalette(range));
+      meshMapper->SetColorModeToMapScalars();
+    }
+    else
+    {
+      // for char, short, int, long arrays 0-255 maps to color component
+      // for float arrays 0-1 maps to color component
+      meshMapper->SetColorModeToDirectScalars();      
+    }
     d->volumeOptions->setSurfaceColored(true);
+    emit contourChanged();
```

Comment:
Why is `countourChanged` only emitted if there are scalars?

Reply:
Good catch. It needs to be emitted always.